gravitee-apim-console-webui/src/management/api/notifications/notifications-list/notifications-list.component.ts

```diff
@@ -13,11 +13,81 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Component } from '@angular/core';
+import { Component, Inject, OnInit } from '@angular/core';
+import { takeUntil, tap } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+
+import { NotificationSettingsService } from '../../../../services-ngx/notification-settings.service';
+import { UIRouterStateParams } from '../../../../ajs-upgraded-providers';
+import { GioTableWrapperFilters } from '../../../../shared/components/gio-table-wrapper/gio-table-wrapper.component';
+import { gioTableFilterCollection } from '../../../../shared/components/gio-table-wrapper/gio-table-wrapper.util';
+
+type NotificationSettingsTable = {
+  name: string;
+  id: string;
+};
+
+export interface NotificationSettingsTableApi {
+  id: string;
+  name: string;
+  referenceType: string;
+  referenceId: string;
+  notifier: string;
+  hooks: [];
+  useSystemProxy: boolean;
+  config_type: string;
+}
 
 @Component({
   selector: 'notifications-list',
   template: require('./notifications-list.component.html'),
   styles: [require('./notifications-list.component.scss')],
 })
-export class NotificationsListComponent {}
+export class NotificationsListComponent implements OnInit {
+  public notificationsSettingsTable: NotificationSettingsTable[] = [];
+  public isLoadingData = true;
+  public displayedColumns = ['name'];
+  public notificationUnpaginatedLength = 0;
+  public filteredNotificationsSettingsTable = [];
+
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    @Inject(UIRouterStateParams) private readonly ajsStateParams,
+    private readonly notificationSettingsService: NotificationSettingsService,
+  ) {}
+
+  public ngOnInit() {
+    this.isLoadingData = true;
+    this.filteredNotificationsSettingsTable = [];
+    this.notificationSettingsService
+      .getNotificationSettings(this.ajsStateParams.apiId)
+      .pipe(
+        tap((settings) => {
+          this.notificationsSettingsTable = settings.map((notificationSettings) => {
+            return {
+              id: notificationSettings.referenceId,
```

Comment:
notificationSettings.id no ?

Reply:
At first I thought the same but apparently "id" is not the obligatory parameter, so couldn't use it

Reply:
ok but referenceId is not the right one too. we can make a call on this subject ;)

Reply:
Done